### PR TITLE
[risk=low][RW-11727] Extraction prototyping

### PR DESCRIPTION
**Pull request by jmthibault79**
---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/config/config_local.json [hunk 1]
```diff
@@ -46,9 +46,9 @@
     "operationalTerraWorkspaceName": "aouwgscohortextraction",
     "operationalTerraWorkspaceBucket": "fc-56d2f6f5-3efa-46f7-8c01-0911fd77f888",
     "extractionMethodConfigurationNamespace": "aouwgscohortextraction-test",
-    "extractionMethodConfigurationName": "GvsExtractCohortFromSampleNames",
-    "extractionMethodConfigurationVersion": 15,
-    "extractionMethodLogicalVersion": 3,
+    "extractionMethodConfigurationName": "joelproto",
```
Comment (jmthibault79) [inline api/config/config_local.json:49]: This is a copy of https://github.com/broadinstitute/gatk/blob/ah_var_store/scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl taken today - looks like commit ae8b6018ee4242d2cdd35da4fd318369a88ddef3

File: api/config/config_local.json [hunk 2]
```diff
@@ -46,9 +46,9 @@
...
+    "extractionMethodConfigurationVersion": 1,
+    "extractionMethodLogicalVersion": 4,
```
Comment (jmthibault79) [inline api/config/config_local.json:51]: Think of this as the Workflow Version, from the RWB perspective.  We have the ability to use this as a discriminator, if we need different Methods for CDRv7 vs CDRv8.

File: api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java [hunk 3]
```diff
@@ -179,7 +179,9 @@ public static class WgsCohortExtractionConfig {
     public Integer extractionMethodLogicalVersion;
     public String extractionDestinationDataset;
     // This should not exceed the value of GenomicExtractionService.MAX_EXTRACTION_SCATTER.
+    // TODO: remove?
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java:182]: the new version chooses a scatter count with an optional override.  We may be able to stop computing this ourselves.

Comment (Qi77Qi) [reply api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java:182]: shall we check with variant's team to see if this needed?

Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java:182]: yes

**Comment by jmthibault79** [comment]
Implemented in #8904 #8905 #8907 #8911 #8914

**Pull request closed without merging**